nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryAPI.java

```diff
@@ -4431,26 +4437,168 @@ public void testLockingBeforeSave() {
     }
 
     @Test
-    public void testRetentionActive() {
+    public void testRecord() {
+        DocumentModel doc = session.createDocumentModel("/", "doc", "File");
+        doc = session.createDocument(doc);
+        DocumentRef docRef = doc.getRef();
+
+        assertFalse(session.isRecord(docRef));
+        assertFalse(doc.isRecord());
+        session.makeRecord(docRef);
+        assertTrue(session.isRecord(docRef));
+        doc.refresh();
+        assertTrue(doc.isRecord());
+
+        // still available when freshly detached
+        doc = session.getDocument(doc.getRef());
+        doc.detach(true);
+        assertTrue(doc.isRecord());
+    }
+
+    @Test
+    public void testRetainUntil() {
         DocumentModel folder = session.createDocumentModel("/", "fold", "Folder");
         folder = session.createDocument(folder);
+        ACP acp = new ACPImpl();
+        ACL acl = new ACLImpl();
+        acl.add(new ACE("bob", "Remove", true));
+        acp.addACL(acl);
+        folder.setACP(acp, true);
```

Comment:
can be replaced by 
```
        ACP acp = new ACPImpl();
        acp.getOrCreateACL().add(new ACE("bob", "Remove", true));
        folder.setACP(acp, true);
```

Reply:
Agreed but again copy/paste from other tests.